src/features/auth/redux/auth.slice.ts

```diff
@@ -0,0 +1,123 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+
+import { LoginRequestDef } from "@app/features/auth/auth";
+
+import { authGetMe, authLogin } from "../api/auth.api";
+import { saveTokens, clearTokens } from "../helpers/auth.helpers";
+import { UserDef } from "../types/auth.types";
+
+export const AUTH_FEATURE_KEY = "auth";
+
+interface InitialStateDef {
+  user: UserDef | null;
+  isAuthenticated: boolean;
+  error: boolean;
+  loading: boolean;
+}
+
+const initialState: InitialStateDef = {
+  user: null,
+  isAuthenticated: false,
+  error: false,
+  loading: false,
+};
+
+export const login = createAsyncThunk(
+  `${AUTH_FEATURE_KEY}/login`,
+  async (values: LoginRequestDef, { rejectWithValue }) => {
+    try {
+      const response = await authLogin(values);
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+export const getMe = createAsyncThunk(
+  `${AUTH_FEATURE_KEY}/getUser`,
+  async (_, { rejectWithValue }) => {
+    try {
+      const response = await authGetMe();
+      return response.data;
+    } catch (err) {
+      return rejectWithValue(err.response.data);
+    }
+  }
+);
+
+const authSlice = createSlice({
+  name: AUTH_FEATURE_KEY,
+  initialState,
+  reducers: {
+    clearUser(state) {
+      state.user = null;
+      state.isAuthenticated = false;
+      clearTokens();
+    },
+  },
+  extraReducers: builder => {
+    /**
+     * LOGIN
+     */
+    builder.addCase(login.pending, state => {
+      state.error = false;
+      state.loading = true;
+    });
+    builder.addCase(login.fulfilled, (state, action) => {
+      /*
+       * expected options from real api
+       * const { IdToken, RefreshToken, ExpiresIn, ...user } = action.payload;
+       *
+       *  limited response from demo api:
+       */
+      const { token } = action.payload;
+
+      state.loading = false;
+      // state.user = null; // user; not implemented
+      state.isAuthenticated = true;
+
+      /* expected condition when real api: if (IdToken && RefreshToken && ExpiresIn) { */
+      if (token) {
+        saveTokens({ token });
+      }
+    });
```

Comment:
I would just simplify this, I think it is very clear that whatever is returned by the login response you would use it here, I don't think we need to make a lot more clear that just some simple code 
```suggestion
      const { token } = action.payload;
      state.loading = false;
      state.isAuthenticated = true;
      if (token) {
        saveTokens({ token });
      }
    });
```